framework/include/image_resource.hpp

```diff
@@ -0,0 +1,277 @@
+#pragma once
+#include <gvk.hpp>
+
+namespace gvk
+{
+	class image_resource_t;
+	class image_resource_impl_t;
+
+	typedef std::unique_ptr<image_resource_t> image_resource;
+
+	// interface of image_resource type, used for abstraction and implementor in bridge pattern
+	class image_resource_base_t
```

Comment:
I fixed this issue and also a few others reported by ReSharper. Should I also add the [[nodiscard]] attribute to functions with a return value as suggested by ReSharper?

Reply:
Personally, I think adding the [[nodiscard]] attribute to most functions is a bit excessive, as only a handful of functions in the standard library have it.

Reply:
That's a good question. I am guilty of not adding `[[nodiscard]]` by myself and have not really made myself familiar with it. So, for now it is okay if you do not add it. (I might add it at some later point in time everywhere reported by ReSharper.)

Reply:
Thought about this a bit more. The standard library uses [[nodiscard]] only for a couple of functions. One important use case is memory allocation: if the result of `new` is discarded, this will result in a memory leak. If the return type of a function is a type that uses RAII, [[nodiscard]] is not needed. The type's destructor will be called when the return value is discarded, thus freeing the resource.
The standard library also uses [[nodiscard]] for `empty()` on containers, but this is apparently to warn programmers who assume that `empty()` empties a container, which is actually done by `clear()`.
So I think the rationale should be: apply [[nodiscard] only if discarding a return value indicates a memory or resource leak, or a common programmer error.

Reply:
Good points! Thanks for investigating!
I have added a section about `[[nodiscard]]` to the [Contribution Guidelines](https://github.com/cg-tuwien/Gears-Vk/blob/master/CONTRIBUTING.md). Feel free to amend or extend the section!